app/assets/javascripts/fortitude/jquery/blocks/modal.js

```diff
@@ -0,0 +1,40 @@
+(function($) {
+  'use strict';
+
+  $(document).on('open.ft.modal', '.modal', function(event) {
+    var $this = $(this);
+    
+    $.screenLock(true);
+    $('.shade').trigger('open.ft.shade');
+    $('.container--fixed-top').css({
+      paddingRight: $.measureScrollBar()
+    });
+    $this.addClass('modal--is-active');
```

Comment:
need `$this.data().showClass` and `$this.data().hideClass` that use a simpler version of waitForAnimation so it only hides/shows if the animation is finished.